### PR TITLE
pyinfra: add page

### PR DESCRIPTION
**Pull request by CrimsonTome**
- **Version of the command being documented (if known):**  2.3

- [x] The page(s) are in the correct platform directories: `common`, `linux`, `osx`, `windows`, `sunos`, `android`, etc.
- [x] The page(s) have at most 8 examples.
- [x] The page description(s) have links to documentation or a homepage.
- [x] The page(s) follow the [content guidelines](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#guidelines).
- [x] The PR title conforms to the recommended [templates](/tldr-pages/tldr/blob/main/CONTRIBUTING.md#commit-message).

### CODE REVIEW

**Comment by CrimsonTome** [comment]
With regards to the 'or' sections in pyinfra.md, I wasn't sure how to format them because they detail how to do the same operation, just with a slightly different syntax - that has been explained at the top. I just thought it would be a little pointless adding a new section for each way of doing things when it has been explained if that makes sense. Any feedback is much appreciated! :smile:

**Comment by marchersimon** [comment]
To keep our pages maintainable we have to enforce some strict rules, for example that each example is isolated from the others. In this case we should probably decide for one of the two versions - whichever you think is more useful.

File: pages/common/pyinfra.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# pyinfra
+
+> Automates infrastructure at a large scale.
+> More information: <https://docs.pyinfra.com>.
+
+- Execute a command over ssh:
```
Comment (marchersimon) [inline pages/common/pyinfra.md:6]:
```suggestion
- Execute a command over `ssh`:
```

File: pages/common/pyinfra.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+`pyinfra {{target_ip_address}} exec -- {{command_name_and_arguments}}`
+
+- Execute contents of a deploy file on a list of targets:
+
+`pyinfra {{path/to/inventory.py}} {{path/to/deploy.py}}`
```
Comment (marchersimon) [inline pages/common/pyinfra.md:12]: I'm not sure if it makes sense here, but it's a bit unclear what `inventory.py` is.
```suggestion
`pyinfra {{path/to/target_list.py}} {{path/to/deploy.py}}`
```

File: pages/common/pyinfra.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+- Execute commands on your local machine:
```
Comment (marchersimon) [inline pages/common/pyinfra.md:14]:
```suggestion
- Execute commands locally:
```
We try to avoid pronouns like `your`.

**Comment by CrimsonTome** [comment]
Thank you for your review @marchersimon. I will adapt the page later on in the day when I have access to a PC. 

inventory.py is the canonical name of the targets list though I believe you can name it what you like.

File: pages/common/pyinfra.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+
+`pyinfra @local exec -- {{command_name_and_arguemnts}}`
+
+or
```
Comment (sbrl) [inline pages/common/pyinfra.md:18]: We have a linter which enforces a strict format, so unfortunately we can't have an `or` branch here. The format is very much
```markdown
- Example description

`example command`
```

File: pages/common/pyinfra.md [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+
+`pyinfra @local {{path/to/deploy.py}}`
+
+- Execute commands over Docker:
+
+`pyinfra @docker/{{container}} exec -- {{command_name_and_arguemnts}}`
+
+or
```
Comment (sbrl) [inline pages/common/pyinfra.md:26]: Same here.

File: pages/common/pyinfra.md [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+
+`pyinfra @local exec -- {{command_name_and_arguemnts}}`
+
+or
+
+`pyinfra @local {{path/to/deploy.py}}`
+
+- Execute commands over Docker:
+
+`pyinfra @docker/{{container}} exec -- {{command_name_and_arguemnts}}`
```
Comment (sbrl) [inline pages/common/pyinfra.md:24]: typo?
```suggestion
`pyinfra @docker/{{container}} exec -- {{command_name_and_arguments}}`
```

**Review by sbrl — COMMENTED** [review]
Disclaimer: I know @CrimsonTome IRL,

Thanks for the new page! I've left some comments below for you to review.

**Comment by CrimsonTome** [comment]
Thank you for the review @sbrl. 

Yeah, I will fix the `or` statements with the preferable option. And oops ty for spotting the typo 😅.
As per my previous comment, I will resolve these later today.

File: pages/common/pyinfra.md (hunk 1 above)
Comment (CleanMachine1) [reply pages/common/pyinfra.md:6]:
```suggestion
- Execute a command over `SSH`:
```

Comment (sbrl) [reply pages/common/pyinfra.md]: This doesn't make any sense to have SSH in backticks, because SSH is the name of a protocol and not a command.

**Review by sbrl — COMMENTED** [review]
Looks ok to me, though as I've commented above SSH does not make sense to be in backticks as it's the name of a protocol and not the name of a command.

**Review by sbrl — APPROVED** [review]

**Review by CleanMachine1 — APPROVED** [review]
Thanks for your contribution!

File: pages/common/pyinfra.md [hunk 7]
```diff
@@ -0,0 +1,20 @@
+# pyinfra
+
+> Automates infrastructure at a large scale.
+> More information: <https://docs.pyinfra.com>.
+
+- Execute a command over `SSH`:
```
Comment (marchersimon) [inline pages/common/pyinfra.md:6]:
```suggestion
- Execute a command over SSH:
```

**Comment by CrimsonTome** [comment]
Okay, all issues *should* be sorted now :)

**Review by navarroaxel — APPROVED** [review]

**Pull request merged**